sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -390,6 +391,171 @@ abstract class UnixTimeExprMeta[A <: BinaryExpression with TimeZoneAwareExpressi
   }
 }
 
+trait GpuNumberToTimestampUnaryExpression extends GpuUnaryExpression {
+  
+  override def dataType: DataType = TimestampType
+  override def outputTypeOverride: DType = DType.TIMESTAMP_MICROSECONDS
+
+  /**
+   * Test whether if input * multiplier will cause Long-overflow. In Math.multiplyExact, 
+   * if there is an integer-overflow, then it will throw an ArithmeticException "long overflow"
+   */
+  def checkLongMultiplicationOverflow(input: ColumnVector, multiplier: Long): Unit = {
+      withResource(input.max()) { maxValue =>
+        Math.multiplyExact(maxValue.getLong, multiplier)
+      }
+      withResource(input.min()) { minValue =>
```

Comment:
`minValue` is a scalar, you need to check if it is valid before reading its value.